### ISSUE TITLE
Bug: `sort-classes` Attributes with the same name but within different groups may not trigger linting

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#209]
### Describe the bug

Comparisons between attributes with the same name currently do not trigger linting. 
This means that a class that only has sorting errors between elements with the same name will not trigger linting and will remain unsorted.

### Code example

Configuration

```
 "groups": [
            "static-property",
            "property"
          ]
```

Valid code:
```ts
class Class {
  a
  static a;
}
```

Invalid code:
```ts
class Class {
  a
  static b;
}
```

However, the code below will be sorted correctly because `b` and `a` will trigger linting.
```ts
class Class {
  b
  a
  static a;
}
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.2.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.
